status-im/status-react
# Add an option to run LES on Ropsten as an experimental feature.

#### 1. LES option

Bring a LES option for Ropsten back available is special fleet is used.

To be more clear that it it there, it is always visible but returns a message if the current fleet doesn't support LES.

#### 2. New fleet

One additional fleet is added to a separate JSON file, having bootnodes with LES servers synced to Ropsten. Separate file is used because while experimenting, we don't want fleets.status.im to return this fleet, and I don't want this file to be accidentally overwritten.

#### 3. A warning message

When switching to a LES network, the app shows a warning message, that this feature is experimental.

#### How to test?

(1) testing LES option
     a. Go to account, enable dev mode, go to the network selection
     b. Observe the list of networks. Ropsten (LES) should be in the list, but when clicking on it there should be a message that LES isn't supported on the current fleet.

(2) a new fleet + warning
     a. go to account → fleets
     b. observe a new fleet `les.dev.ropsten` there, switch to it
     c. go to the network selection, again, click on LES
     d. observe the warning dialog, but the "connect" button appears anyway.
     e. press "connect", you are now using LES!

**NB!** LES will probably not work normally (balance can be always 0, DApps might be broken), that wasn't a scope of this PR to fix all these issues. We need this option to start testing LES.

fixes #5643

status: ready
